### ISSUE TITLE
Support Basic-Auth and custom HTTP Headers when loading a swagger spec

### ISSUE THREAD

**Issue by davidalpert** [go-swagger/go-swagger#2263]
## Problem statement

I am trying to use the go-swagger library to generate a golang client library for a swagger document which is protected and requires:

- a basic auth header
- a custom, app-specific header

My swagger generate client command is failing with this error

````
could not access document at %q [%s]
````

Which I have traced to this swag package

- https://github.com/go-openapi/swag/blob/v0.19.8/loading.go#L57-L62

I have opened https://github.com/go-openapi/swag/issues/44 and https://github.com/go-openapi/swag/pull/45 to make the `swag.LoadFromFileOrHTTP(path string)` method make use of a few new package-level globals which can be used to customize behaviour of the `http.Client{}` used to load the remote swagger spec:

- `swag.LoadHTTPBasicAuthUsername` and `swag.LoadHTTPBasicAuthPassword`, when set, enable basic auth on the swag.loader`http.Client{}`; and
- `swag.LoadHTTPCustomHeaders` is a `map[string]string` which is used to populate HTTP Headers on the swag.loader `http.Client{}`

I propose adding a pair of flags to `generator.GenOpts` and related CLI args to expose these settings to `go-swagger` consumers:
```
// GenOpts the options for the generator
type GenOpts struct {
        ...
	LoadTimeoutSeconds         int32
	LoadBasicAuth              string // "username:password"
	LoadHeaders                string // ["Key1:Value","Key2:value2"]
        ...
}
```

## Steps to reproduce

Run a `swagger generate` command with an `-f` argument pointing at at swagger JSON spec behind basic auth or served by a server requiring custom HTTP headers to be present before responding.

## Environment
swagger version: v0.23.0  
go version: 1.13.3 
OS:   darwin/amd64

**Comment by davidalpert** [comment]
The related changes to `go-openapi/swag` library were released in version `v0.19.9`
- https://github.com/go-openapi/swag/compare/v0.19.8...v0.19.9

**Comment by fredbi** [comment]
fixed with go-openapi/swag#45
